### DISCUSSION TITLE
Cardinal with BoredBrain OPTX and Ableton?

### DISCUSSION THREAD

**Discussion by Mr-V1ct0r** [DISTRHO/Cardinal#213] [Q&A]
I was about to purchase VCV Rack Pro but then I found Aria Salvatrice's story online and decided not to support the company.  Through Aria's Youtube page, I was made aware of Cardinal which was a pleasant surprise.  I installed and VST3 doesn't work (which I was aware could be an issue from reading all the info here on Github).  VST2 works but I can't seem to find a way to route cv to my OPTX module with the Host CV module.  Is there a way to do this in Cardinal?  I've seen it done on VCV Rack Pro demos on Youtube which is one of the reasons why I bought the OPTX.  If not, is it in the future plans?  Thanks in advance.

**Comment by dromer** [comment]
VST2 does not have CV ports. This is well documented.
And most VST3 hosts don't even do the CV implementation.

LV2 and Jack standalone provide a CV interface with "meaningful values" like voltage levels and ranges, instead of just normalized audio frames.
It is implemented in several hosts and run-times, but certainly not Ableton.

You can use Host Params module for controlling the 24+ parameters in our DAW automation. Is there a way for Ableton to convert audio signals to parameter controls?

I am not aware of the `BoredBrain OPTX` device, but in Cardinal we do have the Expert Sleepers encoders https://www.expert-sleepers.co.uk/vcvrack_encoders.html
Does BoredBrain have such a module that integrates with Rack?

[edit:
Oh right, it connects over ADAT, so it really depends on your interface as well.

What do you use to connect this unit to your computer?
And in which OS are you working?

**Comment by Mr-V1ct0r** [comment]
I'm on Windows 10 and it connects with ADAT to my Focusrite Scarlett 18i20.  It just shows up in Ableton as 8 extra audio inputs and 8 extra audio outputs.

EDIT:  The audio outputs are DC coupled.  I have been using Silent Way to send out clock signals as of now.  I wanted to integrate VCV rack to sequence and send modulation to my eurorack setup.

**Comment by falkTX** [comment]
connecting to external hardware from within Cardinal is unsupported. All audio, cv, midi and transport goes through the host, and this is by design.

a way to talk to external hardware is not totally out of question still, but it wont go through the usual host-audio module, instead it could be its own dedicated module.
in any case, no such module exists like that in cardinal today.

**Reply by Mr-V1ct0r** [reply]
So does that mean that Cardinal won't work with the ES-3 and ES-6 either?

**Reply by LKBorg** [reply]
Hej, I have the same issue with the Expert Sleepers ES-3/6 combo 

All works fine in Ableton itself with the CV tools, however Cardinal is limited by the only 8 audio I/O (I connect via ADAT through channels 13-18) and the VST3 version of Cardinal doesn’t seem to present the CV module to Ableton. 

I can route in a CV signal through the Audio 8 host module, but that’s it. 

Really like to use Cardinal with my little Eurorack to extent its functionality! 

cheers
